tests/generation/test_utils.py

```diff
@@ -1021,6 +1022,26 @@ def test_beam_search_generate_dict_outputs_use_cache(self):
                     output, input_ids, model.config, use_cache=True, num_return_sequences=beam_scorer.num_beams
                 )
 
+    @require_accelerate
+    @require_torch_multi_gpu
+    def test_model_parallel_beam_search(self):
+        for model_class in self.all_generative_model_classes:
+            if model_class._no_split_modules is None:
+                continue
+
+            config, input_ids, attention_mask, max_length = self._get_input_ids_and_config()
+
+            model = model_class(config).eval()
+            with tempfile.TemporaryDirectory() as tmp_dir:
+                model.cpu().save_pretrained(tmp_dir)
+                new_model = model_class.from_pretrained(tmp_dir, device_map="auto")
+
+                new_model.generate(
+                    input_ids,
+                    attention_mask=attention_mask,
+                    max_length=max_length,
```

Comment:
I believe this has to be set to invoke beam search. Even if not - it'd be good to be explicit and keep the number of beams small in case defaults change upstream

```suggestion
                    max_length=max_length,
                    num_beams=2,
```

Reply:
Thanks for point it out!
I've tested with greedy search so far 😅